mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,651 @@
+        y_pred_bin = np.where(y_pred == positive_class, 1, 0)
+
+    if y_probs is not None:
+        y_probs = np.array(y_probs)
+        y_prob_bin = y_probs[:, positive_class_index]
+
+    return y_bin, y_pred_bin, y_prob_bin
+
+
+def _get_classifier_per_class_metrics(y, y_pred):
+    """
+    get classifier metrics which computing over a specific class.
+    For binary classifier, y/y_pred is for the positive class.
+    For multiclass classifier, y/y_pred sum up to a binary "is class" and "is not class".
+    """
+    metrics = {}
+    confusion_matrix = sk_metrics.confusion_matrix(y, y_pred)
+    tn, fp, fn, tp = confusion_matrix.ravel()
+    metrics["true_negatives"] = tn
+    metrics["false_positives"] = fp
+    metrics["false_negatives"] = fn
+    metrics["true_positives"] = tp
+    metrics["recall"] = sk_metrics.recall_score(y, y_pred)
+    metrics["precision"] = sk_metrics.precision_score(y, y_pred)
+    metrics["f1_score"] = sk_metrics.f1_score(y, y_pred)
+    return metrics
+
+
+def _get_classifier_global_metrics(is_binomial, y, y_pred, y_probs, labels):
+    """
+    get classifier metrics which computing over all classes examples.
+    """
+    metrics = {}
+    metrics["accuracy"] = sk_metrics.accuracy_score(y, y_pred)
+    metrics["example_count"] = len(y)
+
+    if not is_binomial:
+        metrics["f1_score_micro"] = sk_metrics.f1_score(y, y_pred, average="micro", labels=labels)
+        metrics["f1_score_macro"] = sk_metrics.f1_score(y, y_pred, average="macro", labels=labels)
+
+    if y_probs is not None:
+        metrics["log_loss"] = sk_metrics.log_loss(y, y_probs, labels=labels)
+
+    return metrics
+
+
+def _get_classifier_per_class_metrics_collection_df(y, y_pred, labels):
+    per_class_metrics_list = []
+    for positive_class_index, positive_class in enumerate(labels):
+        (y_bin, y_pred_bin, _,) = _get_binary_sum_up_label_pred_prob(
+            positive_class_index, positive_class, y, y_pred, None
+        )
+
+        per_class_metrics = {"positive_class": positive_class}
+        per_class_metrics.update(_get_classifier_per_class_metrics(y_bin, y_pred_bin))
+        per_class_metrics_list.append(per_class_metrics)
+
+    return pd.DataFrame(per_class_metrics_list)
+
+
+_Curve = namedtuple("_Curve", ["plot_fn", "plot_fn_args", "auc"])
+
+
+def _gen_classifier_curve(
+    is_binomial,
+    y,
+    y_probs,
+    labels,
+    curve_type,
+):
+    """
+    Generate precision-recall curve or ROC curve for classifier.
+    :param is_binomial: True if it is binary classifier otherwise False
+    :param y: True label values
+    :param y_probs: if binary classifer, the predicted probability for positive class.
+                    if multiclass classiifer, the predicted probabilities for all classes.
+    :param labels: The set of labels.
+    :param curve_type: "pr" or "roc"
+    :return: An instance of "_Curve" which includes attributes "plot_fn", "plot_fn_args", "auc".
+    """
+    if curve_type == "roc":
```

Comment:
For both PR and ROC plots, can we plot a dashed greyed-out transparent line that extends to show the base random selection probability? (for PR extending from ((0,1), (1,0)) and for ROC ((0,0), (1,1)) - this greatly helps in visualizing these plots as a 'departure from random chance' for determining how solid the binary classifier is.